src/test/javascript/spec/component/account/vcs-access-token-settings.component.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { MockNgbModalService } from '../../helpers/mocks/service/mock-ngb-modal.service';
+import { MockTranslateService, TranslatePipeMock } from '../../helpers/mocks/service/mock-translate.service';
+import { TranslateService } from '@ngx-translate/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { PROFILE_LOCALVC } from 'app/app.constants';
+import { VcsAccessTokensSettingsComponent } from 'app/shared/user-settings/vcs-access-tokens-settings/vcs-access-tokens-settings.component';
+
+describe('VcsAccessTokensSettingsComponent', () => {
+    let fixture: ComponentFixture<VcsAccessTokensSettingsComponent>;
+    let comp: VcsAccessTokensSettingsComponent;
+
+    let accountServiceMock: { getAuthenticationState: jest.Mock; deleteUserVcsAccessToken: jest.Mock; addNewVcsAccessToken: jest.Mock };
+    let profileServiceMock: { getProfileInfo: jest.Mock };
+    let translateService: TranslateService;
+
+    const token = 'initial-token';
+
+    beforeEach(async () => {
+        profileServiceMock = {
+            getProfileInfo: jest.fn(),
+        };
+        accountServiceMock = {
+            getAuthenticationState: jest.fn(),
+            deleteUserVcsAccessToken: jest.fn(),
+            addNewVcsAccessToken: jest.fn(),
+        };
+
+        await TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [VcsAccessTokensSettingsComponent, TranslatePipeMock],
+            providers: [
+                { provide: AccountService, useValue: accountServiceMock },
+                { provide: ProfileService, useValue: profileServiceMock },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: NgbModal, useClass: MockNgbModalService },
+            ],
+        }).compileComponents();
+        fixture = TestBed.createComponent(VcsAccessTokensSettingsComponent);
+        comp = fixture.componentInstance;
+        translateService = TestBed.inject(TranslateService);
+        translateService.currentLang = 'en';
+    });
+
+    beforeEach(() => {
+        profileServiceMock.getProfileInfo.mockReturnValue(of({ activeProfiles: [PROFILE_LOCALVC] }));
+        accountServiceMock.getAuthenticationState.mockReturnValue(of({ id: 1, vcsAccessToken: token, vcsAccessTokenExpiryDate: '11:20' } as User));
+        accountServiceMock.addNewVcsAccessToken.mockReturnValue(of({ id: 1, vcsAccessToken: token, vcsAccessTokenExpiryDate: '11:20' } as User));
+        accountServiceMock.deleteUserVcsAccessToken.mockReturnValue(of({}));
+    });
+
+    it('should initialize with localVC profile', async () => {
+        comp.ngOnInit();
+        expect(profileServiceMock.getProfileInfo).toHaveBeenCalled();
+        expect(accountServiceMock.getAuthenticationState).toHaveBeenCalled();
+        expect(comp.localVCEnabled).toBeTrue();
+    });
+
+    it('should initialize with no localVC profile set', async () => {
```

Comment:
Can you check if we really need the `async` here? We do not `await` anything in the test

Reply:
I removed it